src/Features/Blockcore.Features.WebHost/Startup.cs

```diff
@@ -73,15 +68,40 @@ public void ConfigureServices(IServiceCollection services)
                     );
                 });
 
-            // Add framework services.
-            services.AddMvc(options =>
+            if (webHostSettings.EnableUI)
```

Comment:
Brilliant stuff

Reply:
Thanks! I think additional improvements could be made in the code, there might be a bit reflection performed that might not be needed if Web Socket is disabled.